lib/modulesync/pr/github.rb

```diff
@@ -39,8 +46,12 @@ def manage(namespace, module_name, options)
         # We only assign labels to the PR if we've discovered a list > 1. The labels MUST
         # already exist. We DO NOT create missing labels.
         return if pr_labels.empty?
-        $stdout.puts "Attaching the following labels to PR #{pr['number']}: #{pr_labels.join(', ')}"
-        @api.add_labels_to_an_issue(repo_path, pr['number'], pr_labels)
+        if options[:noop]
+          puts "Using no-op. Would attach the following labels to the PR: #{pr_labels.join(', ')}"
```

Comment:
You could return here as well, to avoid the `else` clause next line.

Reply:
I'd love to do that, but this is not easily possible. In the [GitHub pr module](https://github.com/voxpupuli/modulesync/pull/188/files#diff-67ea73e264f66bd81ae8cc64cbbb0d38) the code, which sets the labels, needs to run afterwards and has a separate `:noop` clause. Hence, we can't short-circuit, we need to let the flow hop over the hot code.

Or what would be your concrete suggestion?

Reply:
Somewhat unrelated, but if a PR already existed, `pr` is `nil` in the `else` clause. A fix for that has implications for noop. Either you will `return unless pull_requests.empty?` and that means the noop print at the beginning can also return. The alternative is that you add `pr = pull_requests.first` and the label code should print 'Would ensure labels x,y,z on PR'.

Reply:
Right. Turns out there are quite a lot of those "early exit" cases. I think I know addressed them all.

As a result the implementation for GitHub PRs and GitLab MRs has become almost identical. (The API is different for PR/MR labels; GitLab creates them _with_ the MR, GitHub adds them _after_ creation of the PR.)

I hope we're now ready for merging. Can you all take a look again, please?

Reply:
With the current implementation this is unreachable code. On line 25 there's a return after `if options[:noop]`. That's fine, but it means you can drop this line now.

Reply:
True! I was totally keen on removing that clause and align the both implementations, and in the end I forgot the one and important thing! :laughing:

Thanks for catching that! :100: